LmP support for imx8mp-lpddr4-evk

mike-scott opened the pull request:
- Cleanup of build for SPL when SPL_DM is enabled
- Drop hard-coded configs for USB
- Allow for CONFIG_BOOTCOMMAND to be set in defconfig
- Enable runtime detection of raw sector offsets for secondary boot
- Fix imx8m_usb_power_domain for IMX8MP (to enable USB in SPL)
- Make fuse definition check common across IMX8M (from IMX8MM)

igoropaniuk committed:
[FIO internal] imx8mp_evk: disable use of TCPC for SPL

Disable usage of TCPC from the board file for the SPL build, as
from board/freescale/common/Makefile:

ifndef CONFIG_SPL_BUILD
obj-$(CONFIG_USB_TCPC) += tcpc.o
endif

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

igoropaniuk committed:
[FIO internal] imx8mp_evk: drop hardcoded USB configuration

Drop hardcoded USB configuration from imx8mp-lpddr4-evk header.
This allows for changes via defconfig.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

igoropaniuk committed:
[FIO internal] imx8mp_evk: runtime detection raw sector offsets

Add runtime detection for the next boot image (U-Boot proper FIT) SD/MMC
sector offset.

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

Tim-Anderson committed:
[FIO internal] imx8mp_evk: use common spl_mmc_get_uboot_raw_sector()

This function moved as part of 8mm to mach-imx/spl.c

Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

Tim-Anderson committed:
[FIO internal] imx8mp_evk: Only define CONFIG_BOOTCOMMAND if not set

Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

Tim-Anderson committed:
[FIO internal] imx8m: set the fuses definitions common for all imx8m parts

Signed-off-by: Tim Anderson <tim.anderson@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott committed:
[FIO toup] ARM: imx8m: support IMX8MP in imx8m_usb_power_domain

PGC registers differ between imx8m and imx8mp parts.  Let's add
support for IMX8MP registers to enable USB power domain in SPL.

Signed-off-by: Michael Scott <mike@foundries.io>

igoropaniuk committed:
[FIO internal] imx8mp_evk: disable DM_I2C for SPL

Force disable DM_I2C for the SPL build (workaround)

Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>
Signed-off-by: Michael Scott <mike@foundries.io>

Tim-Anderson reviewed (approved):
LGTM

ricardosalveti commented on arch/arm/mach-imx/fiohab.c:
```diff
@@ -97,7 +97,7 @@ static int fiovb_provisioned(void)
 #define SECURE_FUSE_WORD	(6)
 #define SECURE_FUSE_VALUE	(0x00000002)
-#elif CONFIG_IMX8MM
+#elif CONFIG_IMX8M
```
Are all using the same fuses?

mike-scott replied:
This was a change from @Tim-Anderson and I think it's correct.  But I can double check.

mike-scott replied:
These values are correct for 8MM, 8MP (per SRM: 2021-04) and 8MQ (per RM/SRM: 2020-04)

MrCry0 replied:
> These values are correct for 8MM, 8MP (per SRM: 2021-04) and 8MQ (per RM/SRM: 2020-04)


Could we add that additional info to the commit message? Like
```
[FIO internal] imx8m: set the fuses definitions common

The fuse banks parameters are the same for all known imx8m SoC as per (S)RM.
Use the same fuses definitions for the whole imx8m family.
```

ricardosalveti commented:
@MrCry0 most are internal, but please review as you're handling the upstreaming stuff.

mike-scott commented:
@ricardosalveti @Tim-Anderson debugging this PR as full runs.  Hitting a few issues.

MrCry0 commented on include/configs/imx8mp_evk.h:
```diff
@@ -33,6 +33,7 @@
 #define CONFIG_POWER
 #define CONFIG_POWER_I2C
 #define CONFIG_POWER_PCA9450
+#undef CONFIG_DM_I2C
```
IMHO this commit duplicates the functionality from the `scripts/Makefile.uncmd_spl`: the DM_I2C gets undefined in case of CONFIG_SPL_BUILD && !CONFIG_SPL_DM.
SPL_DM isn't defined for all imx8m* for now.

MrCry0 commented on include/configs/imx8mp_evk.h:
```diff
@@ -165,6 +165,7 @@
 			"fi; " \
 		"fi;\0"
 
+#ifndef CONFIG_BOOTCOMMAND
```
Better move the CONFIG_BOOTCOMMAND to defconfig, similar to the solution for 7ulp which is done and approved in the mainline u-boot (https://patchwork.ozlabs.org/project/uboot/patch/20210826110424.1211911-1-oleksandr.suvorov@foundries.io/) - such solution can be easily upstreamed.

mike-scott committed:
[FIO internal] imx8mp_evk: set CONFIG_SYS_BOOTM_LEN for 64MB

This setting is missing from imx8mp which causes the following error
during fit-image load via bootm:
Error: inflate() returned -5
Image too large: increase CONFIG_SYS_BOOTM_LEN
Must RESET board to recover

Signed-off-by: Michael Scott <mike@foundries.io>

MrCry0 reviewed (approved):
Looks good.
There is a slightly tuned version (https://github.com/foundriesio/u-boot/pull/98), but it's up to you which PR to be merged.
